internal/scheduler/jobdb/job.go

```diff
@@ -58,39 +60,6 @@ func EmptyJob(id string) *Job {
 	return &Job{id: id, runsById: map[uuid.UUID]*JobRun{}}
 }
 
-// NewJob creates a new scheduler job
-func NewJob(
```

Comment:
The biggest change in this PR is that it's now necessary to have a jobDb to create jobs. The benefit of this approach is that the jobDb can store various defaults and cached values, which may be useful in the future. This is like what we do with the nodeDb, where the nodeDb is responsible for creating nodes.